src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -101,18 +112,25 @@ instance Hashable CHash where
     hashWithSalt s (CHash h) = hashWithSalt s h
 
 -- | Client address
-newtype CAddress = CAddress CHash deriving (Show, Eq, Generic, Hashable, Buildable)
+-- @w@ is phantom type and stands for type of item this address belongs to.
+newtype CAddress w = CAddress CHash deriving (Show, Eq, Generic, Hashable, Buildable)
+
+-- | Marks address as belonging to wallet set.
+data WS = WS
+
+-- | Marks address as belonging to account.
+data Acc = Acc
```

Comment:
I am not clear about distinction between `CAddress WS` and `CAddress WS`. How should I interpret this on a high level?